src/hooks/useSubgraphTokenSupplies.ts

```diff
@@ -159,3 +166,232 @@ export const useTokenSuppliesQuery = (
 
   return byDateTokenSupplies;
 };
+
+/**
+ * Fetches TokenSupply records from each of the given subgraph URLs,
+ * returning the combined records grouped by date.
+ *
+ * Only the records belonging to the latest block per date are returned.
+ *
+ * This hook handles paging and returns the completed results up to {earliestDate}.
+ *
+ * @param chartName
+ * @param subgraphUrls
+ * @param baseFilter
+ * @param earliestDate the earliest date to fetch, in YYYY-MM-DD format
+ * @param dateOffset the number of days to fetch in each page/request
+ * @returns Records grouped by date, or null if still fetching
+ */
+export const useTokenSuppliesQueries = (
+  chartName: string,
+  subgraphUrls: SUBGRAPH_URLS | null | undefined,
+  _baseFilter: TokenSupply_Filter,
+  _earliestDate: string | null | undefined,
+  _dateOffset?: number,
+): Map<string, TokenSupply[]> | null => {
+  // Cache these props, as they will be passed to the query for each blockchain, and we don't want to propagate non-changes
+  const [baseFilter, setBaseFilter] = useState(_baseFilter);
+  useEffect(() => {
+    if (_baseFilter == baseFilter) return;
+
+    console.debug(`${chartName}: baseFilter changed to ${JSON.stringify(_baseFilter)}`);
+    setBaseFilter(_baseFilter);
+  }, [_baseFilter, chartName]);
```

Comment:
all these `useStates` & `useEffects` in this function can prob be cleaned up into `useMemos` to achieve what you want